evap/results/exporters.py

```diff
@@ -271,58 +285,118 @@ def write_overall_results(
             self.next_row()
 
             # Same reasoning as above.
-            self.write_empty_row_with_styles(["default"] + ["border_top" if gt1 else "default" for gt1 in count_gt_1])
+            self.write_empty_row_with_styles(
+                ["default", "default"] + ["border_top" if gt1 else "default" for gt1 in count_gt_1]
+            )
+
+    @classmethod
+    def _calculate_display_result(
+        cls, questionnaire_id: int, question: Question, results: OrderedDict[int, list[QuestionResult]]
+    ) -> tuple[float | None, float | None]:
+        values = []
+        count_sum = 0
+        approval_count = 0
+
+        for grade_result in results[questionnaire_id]:
+            if grade_result.question.id != question.id or not RatingResult.has_answers(grade_result):
+                continue
+
+            values.append(grade_result.average * grade_result.count_sum)
+            count_sum += grade_result.count_sum
+            if grade_result.question.is_yes_no_question:
+                approval_count += grade_result.approval_count
+
+        if not values:
+            return None, None
+
+        avg = sum(values) / count_sum
+        if question.is_yes_no_question:
+            percent_approval = approval_count / count_sum if count_sum > 0 else 0
+            return avg, percent_approval
+        return avg, None
+
+    @classmethod
+    def _calculate_display_result_average(
+        cls,
+        evaluations_with_results: list[tuple[Evaluation, OrderedDict[int, list[QuestionResult]]]],
+        questionnaire_id: int,
+        question: Question,
+    ) -> tuple[float | None, float | None]:
+        avg_values = []
+        count_avg = 0
+        avg_approval = []
+        count_approval = 0
+
+        for __, results in evaluations_with_results:
+            if (
+                results.get(questionnaire_id) is None
+            ):  # ignore all results without the questionaire for average calculation
+                continue
```

Comment:
From the code, it is very clear to me that this snippet skips results without a questionnaire. I'd much more appreciate the comment telling me 1. why this can happen in the first place (aren't all results always mapped to some question which needs to be part of some questionaire?) and 2. why we want to skip them